linked_list.py

```diff
@@ -64,22 +72,25 @@ def display(self):
         # iterate though the list
         # add the values of each node
         # to the tuple, then display.
-        output_string = ""
+        output_string = u""
         iter_node = self.head
         while iter_node:
             if isinstance(iter_node.val, str):
-                output_string += "'{}'"  .format(iter_node.val)
+                convert = iter_node.val.decode('utf-8')
+                output_string = u"{}'{}'".format(output_string, convert)
+            elif isinstance(iter_node.val, unicode):
+                output_string = u"{}'{}'".format(output_string, iter_node.val)
             else:
-                output_string += "{}" .format(iter_node.val)
+                output_string = u"{}{}".format(output_string, iter_node.val)
             iter_node = iter_node.next
             if iter_node:
-                output_string += ", "
+                output_string = u"{}, ".format(output_string)
             else:
-                return "({})" .format(output_string)
-        return "()"
+                return u"({})".format(output_string)
+        return u"()"
```

Comment:
Out of curiosity, how much of the complexity of this method would be eliminated by implementing a reasonable `__str__` method (or a reasonable `__unicode__` method) for your `Node` class?